new port: devel/py-decorator4

yzgyyang opened the pull request:
devel/py-decorator needs to be updated to 5.x, while some ports don't
support that in their latest releases, including:

- devel/py-pytest-relaxed
- multimedia/py-moviepy

Updated them to use this devel/py-decorator4 until they officially
support 5.x. No functional changes for these ports.

Approved by:	lwhsu (mentor), portmgr (implicit, no functional change)

lwhsu commented on devel/py-decorator4/Makefile:
```diff
@@ -0,0 +1,22 @@
+LICENSE=	MIT
+LICENSE_FILE=	${WRKSRC}/LICENSE.txt
+
+USES=		python:3.6+
```
`3.6+` is not really needed, but since it's a copy so it's also fine to keep it.

yzgyyang replied:
Noted, I thought the same as well so kept them identical.

lwhsu commented:
> Approved by: lwhsu (mentor), portmgr (implicit, no functional change)

Which part of the patch needs portmgr check?

yzgyyang commented:
> > Approved by: lwhsu (mentor), portmgr (implicit, no functional change)
> 
> Which part of the patch needs portmgr check?

I also updated the two ports mentioned without seeking for maintainer approval - since they are cosmetic changes. I am not sure how to best indicate that in the commit message, so the portmgr part is an attempt. ;)

What do you think? @lwhsu

yzgyyang commented:
Also running build tests for the affected ports:

- devel/py-pytest-relaxed
- multimedia/py-moviepy

https://poudriere.ygy.io/build.html?mastername=130Ramd64-pub&build=2021-05-19_07h52m14s

lwhsu commented:
BTW when adding 

> > > Approved by: lwhsu (mentor), portmgr (implicit, no functional change)
> > 
> > 
> > Which part of the patch needs portmgr check?
> 
> I also updated the two ports mentioned without seeking for maintainer approval - since they are cosmetic changes. I am not sure how to best indicate that in the commit message, so the portmgr part is an attempt. ;)
> 
> What do you think? @lwhsu

Oh, chasing the dependency update is an (undocumented? I guess) allowed update. So no worries.

yzgyyang commented:
@lwhsu Do I need to remove the portmgr part, or commit as-is is fine?

lwhsu reviewed (approved):
Looks good to me. BTW, recently I am thinking that adding this "keeping old version" port should come with a DEPRECATED date, which can be set to a long date in the future, for remind us to revise it when the date approaches and not letting the ports tree have too many old but nobody knows it. It should also safe from being deleted when DEPRECATED date comes but there are still users because one should check the dependency when deleting a port, and we can of course extend the date if it needs.

lwhsu commented:
> @lwhsu Do I need to remove the portmgr part, or commit as-is is fine?

I feel that it's not necessary to include that.

yzgyyang commented:
> > @lwhsu Do I need to remove the portmgr part, or commit as-is is fine?
> 
> I feel that it's not necessary to include that.

Thanks! I have removed that part.

yzgyyang commented:
> Looks good to me. BTW, recently I am thinking that adding this "keeping old version" port should come with a DEPRECATED date, which can be set to a long date in the future, for remind us to revise it when the date approaches and not letting the ports tree have too many old but nobody knows it. It should also safe from being deleted when DEPRECATED date comes but there are still users because one should check the dependency when deleting a port, and we can of course extend the date if it needs.

This is a great idea! I updated the PR and the commit message - please reapprove. ;)

yzgyyang committed:
new port: devel/py-decorator4

devel/py-decorator needs to be updated to 5.x, while some ports don't
support that in their latest releases, including:

- devel/py-pytest-relaxed
- multimedia/py-moviepy

Updated them to use this devel/py-decorator4 until they officially
support 5.x. No functional changes for these ports.

An expiry date has been set for this keeping-old-version port, so we
don't forget to delete it once nothing depends on it.

Approved by:	lwhsu (mentor)

yzgyyang commented:
@lwhsu For context, I fixed the issue mentioned by koobs at https://github.com/yzgyyang/freebsd-ports/commit/4d4cadcf9d489d5e78e160190c92f2c8bc44d307, but later wrongly pushed an older branch and committed that one instead.
